service/src/main/java/bio/terra/workspace/common/utils/AwsUtils.java

```diff
@@ -207,6 +222,67 @@ public static AwsCredentialsProvider createWsmCredentialProvider(
         authentication.getGoogleJwtAudience());
   }
 
+  public static Credentials getAssumeServiceRoleCredentials(
+      AwsConfiguration.Authentication authentication, Environment environment, Duration duration) {
+    AssumeRoleWithWebIdentityRequest request =
+        createRefreshRequest(
+            environment.getWorkspaceManagerRoleArn(),
+            duration,
+            authentication.getGoogleJwtAudience());
+    logger.info(
+        "Assuming Service role ('{}') with session name `{}`, duration {} seconds.",
+        request.roleArn(),
+        request.roleSessionName(),
+        request.durationSeconds());
+
+    return getStsClient().assumeRoleWithWebIdentity(request).credentials();
+  }
+
+  public static Credentials getAssumeUserRoleCredentials(
```

Comment:
This need to pass a container of tags that get consolidated with the tags that the `assumeUserRoleFromServiceCredentials()` method adds.  As this is written, the tags necessary to access the folder cannot be passed, so the resulting credential will not have access to the storage folder.

Reply:
Agree, relates to https://github.com/DataBiosphere/terra-workspace-manager/pull/1203#discussion_r1170295343 (missed adding [principal tags).